src/omp/backend.f90

```diff
@@ -309,43 +305,16 @@ subroutine reorder_omp(self, u_, u, direction)
     class(field_t), intent(inout) :: u_
     class(field_t), intent(in) :: u
     integer, intent(in) :: direction
-    integer :: ndir_loc, ndir_groups
     integer :: i, j, k
     integer :: out_i, out_j, out_k
 
-    select case (direction)
-    case (RDR_X2Y)
-      ndir_loc = self%xdirps%n
-      ndir_groups = self%xdirps%n_blocks
-    case (RDR_X2Z)
-      ndir_loc = self%xdirps%n
-      ndir_groups = self%xdirps%n_blocks
-    case (RDR_Y2X)
-      ndir_loc = self%ydirps%n
-      ndir_groups = self%ydirps%n_blocks
-    case (RDR_Y2Z)
-      ndir_loc = self%ydirps%n
-      ndir_groups = self%ydirps%n_blocks
-    case (RDR_Z2X)
-      ndir_loc = self%zdirps%n
-      ndir_groups = self%zdirps%n_blocks
-    case (RDR_Z2Y)
-      ndir_loc = self%zdirps%n
-      ndir_groups = self%zdirps%n_blocks
-    case default
-      ndir_loc = 0
-      ndir_groups = 0
-      error stop 'unsuported reordering'
-    end select
-
     !$omp parallel do private(out_i, out_j, out_k) collapse(2)
-    do k = 1, ndir_groups
-      do j = 1, ndir_loc
-        do i = 1, SZ
+    do k = 1, u%n_groups
+      do j = 1, u%n
+        do i = 1, u%sz
```

Comment:
To simplify the `reorder_omp` how about introducing a variable `n_groups(3)` in the `allocator_t` class and;
```
do k = 1, self%allocator%n_groups(u%dir)
  do j = 1, self%allocator%cdims(u%dir)
    do i = 1, self%sz (which is from backend%sz or SZ directly)
```
It doesn't require us passing anything extra and removes all the select case stuff above.